File: src/main/java/com/gmail/borlandlp/minigamesdtools/listener/Commands.java
package com.gmail.borlandlp.minigamesdtools.listener;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase;
import com.gmail.borlandlp.minigamesdtools.arena.exceptions.ArenaAlreadyStartedException;
import com.gmail.borlandlp.minigamesdtools.arena.team.TeamProvider;
import com.gmail.borlandlp.minigamesdtools.config.exception.InvalidPathException;
import com.gmail.borlandlp.minigamesdtools.creator.DataProvider;
import com.gmail.borlandlp.minigamesdtools.gui.hotbar.Hotbar;
import com.gmail.borlandlp.minigamesdtools.party.Party;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;


public class Commands implements CommandExecutor {
   public boolean onCommand(CommandSender sender, org.bukkit.command.Command cmd, String label, String[] args) {
      if(cmd.getName().equalsIgnoreCase("arena")) {

          // debug...
          Player player = (Player) sender;
        if(args[0].equalsIgnoreCase("lobby")) {
          MinigamesDTools.getInstance().getLobbyHubAPI().getLobbyByID("spawn_lobby").registerPlayer(player);
        } else if(args[0].equalsIgnoreCase("lobby_leave")) {
          MinigamesDTools.getInstance().getLobbyHubAPI().getLobbyByID("spawn_lobby").unregisterPlayer(player);
        } else if(args[0].equalsIgnoreCase("lobby_transfer")) {
          try {
              MinigamesDTools.getInstance().getLobbyHubAPI().getLobbyByID("example_lobby").transferPlayer(player, MinigamesDTools.getInstance().getLobbyHubAPI().getLobbyByID("example_lobby2"));
          } catch (Exception e) {
              e.printStackTrace();
          }
        } else if (!player.hasPermission("arena.admin")) {
            player.sendMessage(MinigamesDTools.getPrefix() + " Недостаточно прав для доступа.");
            return true;
        } else if (args[0].equalsIgnoreCase("fullreload")) {
              try {
                  MinigamesDTools.getInstance().reload();
              } catch (InvalidPathException e) {
                  e.printStackTrace();
              }
              player.sendMessage(MinigamesDTools.getPrefix() + " Произведена полная перезагрузка плагина.");
           return true;
        } else if(args[0].equalsIgnoreCase("enable")) {
            if(args.length == 2) {
                ArenaBase arena = MinigamesDTools.getInstance().getArenaAPI().getArena(args[1]);
                if(arena != null) {
                    arena.setEnabled(true);
                    player.sendMessage(MinigamesDTools.getPrefix() + " Арена с названием" + args[1] + " включена.");
                } else {
                    player.sendMessage(MinigamesDTools.getPrefix() + " Арена с названием" + args[1] + " не найдена.");
                }
            } else {
                player.sendMessage(MinigamesDTools.getPrefix() + "/arena enable arena_name");
            }
        } else if(args[0].equalsIgnoreCase("disable")) {
            if(args.length == 2) {
                ArenaBase arena = MinigamesDTools.getInstance().getArenaAPI().getArena(args[1]);
                if(arena != null) {
                    arena.forceDisable();
                    player.sendMessage(MinigamesDTools.getPrefix() + " Арена с названием" + args[1] + " выключена.");
                } else {
                    player.sendMessage(MinigamesDTools.getPrefix() + " Арена с названием" + args[1] + " не найдена.");
                }
            } else {
                player.sendMessage(MinigamesDTools.getPrefix() + "/arena enable arena_name");
            }
        } else if(args[0].equalsIgnoreCase("dbg_join") && player.hasPermission("arena.reload")) {

            MinigamesDTools.getInstance().getArenaAPI().arenaJoinRequest(args[1], player);
            player.sendMessage("dbg_join: " + args[1]);
            return true;

        } else if(args[0].equalsIgnoreCase("dbg_state")) {

            ArenaBase.STATE state = ArenaBase.STATE.valueOf(args[2]);
            MinigamesDTools.getInstance().getArenaAPI().getArena(args[1]).setState(state);
            player.sendMessage("dbg_state:" + args[1] + " " + MinigamesDTools.getInstance().getArenaAPI().getArena(args[1]));
            return true;

        } else if(args[0].equalsIgnoreCase("dbg_members")) {
             player.sendMessage("===dbg_teams===");
             for(TeamProvider team : MinigamesDTools.getInstance().getArenaAPI().getArena(args[1]).getTeamController().getTeams()) {
                 player.sendMessage("dbg_teams->" + args[1] + ":" + team.getName() + "#" + team.getPlayers());
             }

             return true;
        } else if(args[0].equalsIgnoreCase("dbg_start")) {
             player.sendMessage("===arena_start===");
            try {
                MinigamesDTools.getInstance().getArenaAPI().getArena(args[1]).startArena();
            } catch (ArenaAlreadyStartedException e) {
                e.printStackTrace();
            }

            return true;
        } else if(args[0].equalsIgnoreCase("hotbar")) {
              try {
                  Hotbar hotbar = MinigamesDTools.getInstance().getHotbarCreatorHub().createHotbar("example_skyhotbar", new DataProvider());
                  MinigamesDTools.getInstance().getHotbarAPI().bindHotbar(hotbar, player);
              } catch (Exception e) {
                  e.printStackTrace();
              }
        } else if(args[0].equalsIgnoreCase("dbg_stop")) {
             player.sendMessage("===arena_stop===");
             player.sendMessage("stopped" + args[1]);
             MinigamesDTools.getInstance().getArenaAPI().getArena(args[1]).forceDisable();

             return true;
        } else if(args[0].equalsIgnoreCase("leave")) {
             MinigamesDTools.getInstance().getArenaAPI().arenaLeaveRequest(player);

             return true;
        } else if(args[0].equalsIgnoreCase("party_create")) {
            Party party = MinigamesDTools.getInstance().getPartyAPI().createParty(player);
            MinigamesDTools.getInstance().getPartyAPI().addParty(party);

            return true;
        } else if(args[0].equalsIgnoreCase("party_join")) {
            MinigamesDTools.getInstance().getPartyAPI().getPartyOf(args[1]).addPlayer(player);
            return true;
        } else if(args[0].equalsIgnoreCase("party_members")) {
            System.out.print(MinigamesDTools.getInstance().getPartyAPI().getPartyOf(player));

            return true;
        } else if(args[0].equalsIgnoreCase("all_arenas")) {
            player.sendMessage("Arenas list:");
            String msg = null;
            for(ArenaBase arena : MinigamesDTools.getInstance().getArenaAPI().getArenas()) {
                msg = " -" + ChatColor.GREEN + arena.getName()
                        + ChatColor.GRAY + " [ID:" + arena.getGameId() + "]"
                        + ChatColor.GREEN + " state:'" + ChatColor.DARK_GREEN + arena.getState() + ChatColor.GREEN + "'"
                        + " players:'"+ ChatColor.DARK_GREEN + arena.getTeamController().countCurrentPlayers()+ ChatColor.GREEN + "'";
                player.sendMessage(msg);
            }
        }

        player.sendMessage("Wrong command");
        return true;
     }

      return false;
   }

}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/listener/Commands.kt
package com.gmail.borlandlp.minigamesdtools.listener

import com.gmail.borlandlp.minigamesdtools.DefaultCreators
import com.gmail.borlandlp.minigamesdtools.MinigamesDTools
import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase
import com.gmail.borlandlp.minigamesdtools.arena.exceptions.ArenaAlreadyStartedException
import com.gmail.borlandlp.minigamesdtools.config.exception.InvalidPathException
import com.gmail.borlandlp.minigamesdtools.creator.DataProvider
import com.gmail.borlandlp.minigamesdtools.gui.hotbar.Hotbar
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class Commands : CommandExecutor {
    override fun onCommand(
        sender: CommandSender,
        cmd: Command,
        label: String,
        args: Array<String>
    ): Boolean {
        if (cmd.name.equals("arena", ignoreCase = true)) { // debug...
            val player = sender as Player
            if (args[0].equals("lobby", ignoreCase = true)) {
                MinigamesDTools.instance!!.lobbyHubAPI!!.getLobbyByID("spawn_lobby")!!.registerPlayer(player)
            } else if (args[0].equals("lobby_leave", ignoreCase = true)) {
                MinigamesDTools.instance!!.lobbyHubAPI!!.getLobbyByID("spawn_lobby")!!.unregisterPlayer(player)
            } else if (args[0].equals("lobby_transfer", ignoreCase = true)) {
                try {
                    MinigamesDTools.instance!!.lobbyHubAPI!!.getLobbyByID("example_lobby")!!.transferPlayer(
                        player,
                        MinigamesDTools.instance!!.lobbyHubAPI!!.getLobbyByID("example_lobby2")!!
                    )
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            } else if (!player.hasPermission("arena.admin")) {
                player.sendMessage(MinigamesDTools.prefix + " Недостаточно прав для доступа.")
                return true
            } else if (args[0].equals("fullreload", ignoreCase = true)) {
                try {
                    MinigamesDTools.instance!!.reload()
                } catch (e: InvalidPathException) {
                    e.printStackTrace()
                }
                player.sendMessage(MinigamesDTools.prefix + " Произведена полная перезагрузка плагина.")
                return true
            } else if (args[0].equals("enable", ignoreCase = true)) {
                if (args.size == 2) {
                    val arena = MinigamesDTools.instance!!.arenaAPI!!.getArena(args[1])
                    if (arena != null) {
                        arena.isEnabled = true
                        player.sendMessage(MinigamesDTools.prefix + " Арена с названием" + args[1] + " включена.")
                    } else {
                        player.sendMessage(MinigamesDTools.prefix + " Арена с названием" + args[1] + " не найдена.")
                    }
                } else {
                    player.sendMessage(MinigamesDTools.prefix + "/arena enable arena_name")
                }
            } else if (args[0].equals("disable", ignoreCase = true)) {
                if (args.size == 2) {
                    val arena = MinigamesDTools.instance!!.arenaAPI!!.getArena(args[1])
                    if (arena != null) {
                        arena.forceDisable()
                        player.sendMessage(MinigamesDTools.prefix + " Арена с названием" + args[1] + " выключена.")
                    } else {
                        player.sendMessage(MinigamesDTools.prefix + " Арена с названием" + args[1] + " не найдена.")
                    }
                } else {
                    player.sendMessage(MinigamesDTools.prefix + "/arena enable arena_name")
                }
            } else if (args[0].equals("dbg_join", ignoreCase = true) && player.hasPermission("arena.reload")) {
                MinigamesDTools.instance!!.arenaAPI!!.arenaJoinRequest(args[1], player)
                player.sendMessage("dbg_join: " + args[1])
                return true
            } else if (args[0].equals("dbg_state", ignoreCase = true)) {
                val state = ArenaBase.STATE.valueOf(args[2])
                MinigamesDTools.instance!!.arenaAPI!!.getArena(args[1])!!.state = state
                player.sendMessage(
                    "dbg_state:" + args[1] + " " + MinigamesDTools.instance!!.arenaAPI!!.getArena(
                        args[1]
                    )
                )
                return true
            } else if (args[0].equals("dbg_members", ignoreCase = true)) {
                player.sendMessage("===dbg_teams===")
                for (team in MinigamesDTools.instance!!.arenaAPI!!.getArena(args[1])!!.teamController!!.teams) {
                    player.sendMessage("dbg_teams->" + args[1] + ":" + team.name + "#" + team.players)
                }
                return true
            } else if (args[0].equals("dbg_start", ignoreCase = true)) {
                player.sendMessage("===arena_start===")
                try {
                    MinigamesDTools.instance!!.arenaAPI!!.getArena(args[1])!!.startArena()
                } catch (e: ArenaAlreadyStartedException) {
                    e.printStackTrace()
                }
                return true
            } else if (args[0].equals("hotbar", ignoreCase = true)) {
                try {
                    MinigamesDTools.instance!!.creatorsRegistry.get(DefaultCreators.HOTBAR.pseudoName)!!
                        .create("example_skyhotbar",
                            DataProvider()
                        ).apply {
                            if (this is Hotbar) {
                                MinigamesDTools.instance!!.hotbarAPI!!.bindHotbar(this, player)
                            }
                        }
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            } else if (args[0].equals("dbg_stop", ignoreCase = true)) {
                player.sendMessage("===arena_stop===")
                player.sendMessage("stopped" + args[1])
                MinigamesDTools.instance!!.arenaAPI!!.getArena(args[1])!!.forceDisable()
                return true
            } else if (args[0].equals("leave", ignoreCase = true)) {
                MinigamesDTools.instance!!.arenaAPI!!.arenaLeaveRequest(player)
                return true
            } else if (args[0].equals("party_create", ignoreCase = true)) {
                val party = MinigamesDTools.instance!!.partyAPI!!.createParty(player)
                MinigamesDTools.instance!!.partyAPI!!.addParty(party)
                return true
            } else if (args[0].equals("party_join", ignoreCase = true)) {
                MinigamesDTools.instance!!.partyAPI!!.getPartyOf(args[1])!!.addPlayer(player)
                return true
            } else if (args[0].equals("party_members", ignoreCase = true)) {
                print(MinigamesDTools.instance!!.partyAPI!!.getPartyOf(player))
                return true
            } else if (args[0].equals("all_arenas", ignoreCase = true)) {
                player.sendMessage("Arenas list:")
                var msg: String?
                for (arena in MinigamesDTools.instance!!.arenaAPI!!.arenas) {
                    msg = (" -" + ChatColor.GREEN + arena.name
                            + ChatColor.GRAY + " [ID:" + arena.gameId + "]"
                            + ChatColor.GREEN + " state:'" + ChatColor.DARK_GREEN + arena.state + ChatColor.GREEN + "'"
                            + " players:'" + ChatColor.DARK_GREEN + arena.teamController!!.countCurrentPlayers() + ChatColor.GREEN + "'")
                    player.sendMessage(msg)
                }
            }
            player.sendMessage("Wrong command")
            return true
        }
        return false
    }
}